docs/src/aleph_bft_api.md

```diff
@@ -60,11 +70,11 @@ A typical implementation of KeyBox would be a collection of `N` public keys, an
 
 ### 3.2 Examples
 
-While the implementations of `KeyBox` and `Network` are pretty much universal, the implementation of `DataIO` depends on the specific application. We consider two examples here.
+While the implementations of `KeyBox` and `Network` are pretty much universal, the implementation of `DataProvider` and `FinalizationProvider` depends on the specific application. We consider two examples here.
 
 #### 3.2.1 Blockchain Finality Gadget.
 
-Consider a blockchain that does not have an intrinsic finality mechanism, so for instance it might be a PoW chain with probabilistic finality or a chain based on PoS that uses some probabilistic or round-robin block proposal mechanism. Each of the `N` nodes in the network has its own view of the chain and at certain moments in time there might be conflicts on what the given nodes consider as the "tip" of the blockchain (because of possible forks or network delays). We would like to design a finality mechanism based on AlephBFT that will allow the nodes to have agreement on what the "tip" is. Towards this end we just need to implement a suitable `DataIO` object and filtering of network messages for AlephBFT.
+Consider a blockchain that does not have an intrinsic finality mechanism, so for instance it might be a PoW chain with probabilistic finality or a chain based on PoS that uses some probabilistic or round-robin block proposal mechanism. Each of the `N` nodes in the network has its own view of the chain and at certain moments in time there might be conflicts on what the given nodes consider as the "tip" of the blockchain (because of possible forks or network delays). We would like to design a finality mechanism based on AlephBFT that will allow the nodes to have agreement on what the "tip" is. Towards this end we just need to implement a suitable `DataProvider` object and filtering of network messages for AlephBFT.
```

Comment:
The description below still uses `send_ordered_batch` rather than the new interface.

Reply:
fixed